### ISSUE TITLE
Querying properties that contain #

### ISSUE THREAD

**Issue by fduser1** [P0lip/nimma#4]
Tried to query the following JSON document with the JSONPath `$..type^[*]^`, 
```json
{
    "properties": {
        "test#1": {
            "type": "string"
        }
    }
}
```

I encounted the error: _TypeError: Cannot read property 'toString' of undefined_.  I noticed the error does not occur if property name does not include #.

For demonstration purposes, I made my query (i.e. [*]) trivial here. In my real use-case, I was using a similar JSONPath query with a non-trivial filter [?(...)] in the Spectral context.

**Comment by P0lip** [comment]
I'll try to look into it this week
I'm running a bit behind due to the vacation I was on for the past 2 weeks.

**Comment by fduser1** [comment]
Welcome back! Thanks for the update.

**Comment by P0lip** [comment]
Hey!
Apologies for such a delay.
I've looked into it now, and I've noticed you make use of the parent operator - `^`, which `nimma` provides only partial support for.
`^` is an additional specific to jsonpath-plus. `nimma` supports `^` only when placed at the end of the entire expression.
For greater compatibility, it offers a fallback to jsonpath-plus to cover off such cases. The error you see there is indeed raised by jsonpath-plus, thus I'd suggest filing an issue in that repo.
